source/tutorials/kubernetes.rst

```diff
@@ -3,354 +3,716 @@
+init` command.
 
-         Swap will be enabled at next reboot, causing failures in
-         your cluster if you choose this method.
 
-   or:
+Choose a pod network add-on
+===========================
 
-   b) Permanently:
+See `What is a Container Network Interface (CNI)?`_ for information on what
+pod network add-ons and CNIs.
 
-      Mask the swap partition:
+It is important to decide which CNI will be used early because some pod
+network add-ons require configuration during cluster initialization. Check
+whether or not your add-on requires special flags when you initialize the
+master control plane.
 
-      .. code-block:: bash
+If your chosen network add-on requires appending to the :command:`kubeadm
+init` command, make note of it before continuing. For example, if you choose
+the *flannel* pod network add-on, then in later steps you must add the
+following to the :command:`kubeadm init` command:
 
-         sudo systemctl mask $(sed -n -e 's#^/dev/\([0-9a-z]*\).*#dev-\1.swap#p' /proc/swaps) 2>/dev/null
-         sudo swapoff -a
+.. code-block:: console
+
+   --pod-network-cidr 10.244.0.0/16
 
-      .. note::
 
-         On systems with limited resources, some performance degradation may
-         be observed while swap is disabled.
 
-#. Switch to root to modify the `hosts` file:
+Choose a container runtime
+==========================
+
+See `What is a Container Runtime Interface (CRI)?`_ for more information on
+what a CRI is.
+
+|CL| supports Kubernetes with the various runtimes
+below with or without `Kata Containers`_:
+
+* `CRI+O`_
+* `containerd`_
+* `Docker`_
+
+The container runtime that you choose will dictate the steps necessary to
+initialize the master cluster with :command:`kubeadm init`.
+
+CRI+O
+-----
+
+For information on CRI+O as a Kubernetes container runtime, see `What is
+CRI+O?`_. To use CRI+O as the Kubernetes container runtime:
+
+#. Start the CRI-O service and enable it to run at boot automatically:
 
    .. code-block:: bash
 
-      sudo -s
+      sudo systemctl enable --now crio.service
 
-#.  Create (or edit if it exists) the hosts file that Kubernetes will read to
-    locate the master's host:
+   When the crio service starts for the first time, it will create a
+   configuration file for crio at :file:`/etc/crio/crio.conf`.
 
-    .. code-block:: bash
+#. Run the kubeadm command to initialize the master node with the
+   :command:`--cri-socket` parameter:
 
-       echo "127.0.0.1 localhost `hostname`" >> /etc/hosts
+   .. important:: 
 
-#.  Exit root:
+      You may need to add additional parameters to the command below,
+      depending the pod network addon in use. 
+      
+      In this example, the :command:`--pod-network-cidr 10.244.0.0/16`
+      parameter is to use *flannel* as the pod networking. See `Choose a pod
+      network add-on`_ for more information.
 
-    .. code-block:: bash
+   .. code-block:: bash
 
-       exit
+      sudo kubeadm init \
```

Comment:
Did this actually work? I had to do the following for CRIO. I saw you had this in the containerd section. 
```
sudo mkdir -p /etc/systemd/system/kubelet.service.d/
cat << EOF | sudo tee  /etc/systemd/system/kubelet.service.d/10-cgroup-driver.conf
[Service]
Environment="KUBELET_EXTRA_ARGS=--cgroup-driver=systemd"
EOF
sudo systemctl daemon-reload
```